cocos/bindings/dop/BufferAllocator.h

```diff
@@ -37,42 +37,58 @@ namespace se {
 class CC_DLL BufferAllocator final : public cc::Object {
 public:
     template <class T>
-    CC_INLINE static T *getBuffer(PoolType type, uint index) {
-        uint size = 0;
-        return BufferAllocator::getBuffer<T>(type, index, &size);
+    static T *getBuffer(PoolType type, uint index) {
+        index &= _bufferMask;
+        const auto p = GET_ARRAY_POOL_ID(type);
+
+#ifdef CC_DEBUG
+        CCASSERT(BufferAllocator::_pools[p] != nullptr, "BufferPool: Invalid buffer pool type");
+#endif
+
+        const auto pool = BufferAllocator::_pools[p];
+
+#ifdef CC_DEBUG
+        CCASSERT(pool->_caches.count(index) != 0, "BufferPool: Invalid buffer pool index");
+#endif
+
+        return reinterpret_cast<T *>(pool->_caches[index]);
     }
 
     template <class T>
     static T *getBuffer(PoolType type, uint index, uint *size) {
         index &= _bufferMask;
-        if (BufferAllocator::_pools.count(type) != 0) {
-            const auto pool = BufferAllocator::_pools[type];
-            if (pool->_buffers.count(index) != 0) {
-                T *ret = nullptr;
-                size_t len;
-                pool->_buffers[index]->getArrayBufferData((uint8_t **)&ret, &len);
-                *size = (uint)len;
-                return ret;
-            } else {
-                return nullptr;
-            }
-        } else {
-            return nullptr;
-        }
+        const auto p = GET_ARRAY_POOL_ID(type);
+
+#ifdef CC_DEBUG
+        CCASSERT(BufferAllocator::_pools[p] != nullptr, "BufferPool: Invalid buffer pool type");
+#endif
+
+        const auto pool = BufferAllocator::_pools[p];
+
+#ifdef CC_DEBUG
+        CCASSERT(pool->_buffers.count(index) != 0, "BufferPool: Invalid buffer pool index");
+#endif
+
+        T *    ret = nullptr;
+        size_t len;
+        pool->_buffers[index]->getArrayBufferData((uint8_t **)&ret, &len);
+        *size = (uint)len;
+        return ret;
     }
 
     BufferAllocator(PoolType type);
     ~BufferAllocator();
 
     Object *alloc(uint index, uint bytes);
-    void free(uint index);
+    void    free(uint index);
 
 private:
-    static cc::map<PoolType, BufferAllocator *> _pools;
-    static constexpr uint _bufferMask = ~(1 << 30);
+    static cc::vector<BufferAllocator *> _pools;
```

Comment:
Without pool type information, how to find correct pool?

Reply:
> Without pool type information, how to find correct pool?

PoolType is continuous, and several different pools are indexed from 0 and then converted to id values based on type.